src/status_im/chat/views/message/message.cljs

```diff
@@ -111,13 +114,39 @@
        ffirst
        replacements))
 
+(defn- parse-url [string on-press]
+  (if (string? string)
+    (let [general-text  (string/split string regx-url)
+          general-text' (if (zero? (count general-text))
+                          [nil]
+                          general-text)
+          link-text     (vec (map-indexed (fn [idx url]
+                                            (let [style {:color colors/blue}]
+                                              [react/text
+                                               {:key      idx
+                                                :style    style
+                                                :on-press #(on-press url)}
+                                               url]))
+                                          (re-seq regx-url string)))
+          link-text'    (if (> (count general-text)
+                               (count link-text))
+                          (conj link-text nil)
+                          link-text)]
+      (mapcat vector general-text' link-text'))
+    string))
+
 ;; todo rewrite this, naive implementation
-(defn- parse-text [string]
+(defn- parse-text [string url-on-press]
   (if (string? string)
     (let [general-text  (string/split string regx)
           general-text' (if (zero? (count general-text))
                           [nil]
                           general-text)
+          general-text' (vec (map-indexed (fn [idx string]
```

Comment:
Can we came up with a more descriptive var name?
Also let's extract this logic as a private function.

Reply:
I'm also not big fan of `symbol'` etc. I would rather have one binding with combined expression (using `->`/`->>` threading macros), or more descriptive names (such as `general-text-seq`,  `general-text-react-seq`, describing each step of the transformation process)